sympy/integrals/rubi/rubi.py

```diff
@@ -305,12 +305,13 @@ def rubi_integrate(expr, var, showsteps=False):
 
     Returns Integral object if unable to integrate.
     '''
+    expr = S(expr)
+    if isinstance(expr, (int, Integer)) or isinstance(expr, (float, Float)) or expr.as_independent(var)[1] == (0 or 1):
```

Comment:
A better way is to check if the variable is in `expr.free_symbols`. That will cover all constant cases.

Reply:
but u have to try constant alphabet just as any alphabet(p) for that it may be show error